### ISSUE TITLE
Check if file exists prior to attempting to delete it

### ISSUE THREAD

**Issue by sentientsolutions** [mikehaertl/phpwkhtmltopdf#353]
Hey!

We regularly notice

`[14-Jul-2020 16:22:24 Europe/Dublin] PHP Warning:  unlink(E:\temp\tmp22ED.tmp.html): No such file or directory in E:\vendor\mikehaertl\php-tmpfile\src\File.php on line 60`

Appearing in our logs. I had a look and line 60 and it's surroundings are:

```php
/**
 * Delete tmp file on shutdown if `$delete` is `true`
 */
 public function __destruct()
 {
    if ($this->delete) {
        unlink($this->_fileName);
    }
 }
```

Can I suggest modifying it slightly to check if the file actually still exists before attempting to delete it? file_exists should work nicely.

```php
/**
 * Delete tmp file on shutdown if `$delete` is `true`
 */
 public function __destruct()
 {
    if ($this->delete && file_exists($this->_fileName)) {
        unlink($this->_fileName);
    }
 }
```

**Comment by mikehaertl** [comment]
Hmm, that's weird. I never had this issue and had no reports like this so far. The file is created and then deleted only here, so how can it not exist? It would be great if you could first debug this further, e.g. where else is the file deleted? Or does this maybe only happen under certain (error?) conditions? 

If you can provide some more detail or even a short example to reliably reproduce it I can open an issue over at [php-tmpfile](https://github.com/mikehaertl/php-tmpfile).

**Comment by sentientsolutions** [comment]
Ok, let me dig a little further and I'll come back to you. 

It's definitely not a 1-to-1 issue. PDFs are being generated en mass by our users without the error appearing in the logs. It's definitely the exception and not the rule. Sadly it doesn't come up following any other errors either.

Let me play around with it and see what I come up with.

**Comment by sentientsolutions** [comment]
Hey Mike.

So as an update. I went and changed the folder that the temp files from the default system temp folder, to a totally separate folder just for the PDF generation script. This seems to have caused the problem to go away, as it has totally stopped coming into the logs.

At a guess I would imagine some script on IIS is performing some sort of garbage collection in the default temp folder? If so, it might be worth the addition of the extra file_exists($this->_fileName).

**Comment by mikehaertl** [comment]
Ok, I'm convinced. By it's nature a temporary directory is not guaranteed to not being deleted from some other process. And the system temp directory is the default. So I'll add the check.

**Comment by mikehaertl** [comment]
@sentientsolutions I've just released 1.1.5 of php-tmpfile which contains this fix. Thanks for your help.
